src/mbgl/style/parser.hpp

```diff
@@ -27,7 +27,7 @@ class Parser {
 
     StyleParseResult parse(const std::string&);
 
-    std::string spriteURL;
+    std::vector<std::unique_ptr<Sprite>> sprites;
```

Comment:
I think using unique pointers but then using `.get()` to get a raw pointer is dangerous because now someone might hold onto a dangling pointer.

I would try to avoid using raw pointers as much as possible, in this case since `Sprite` is such a lightweight object copying is probably even OK.

Reply:
I rewrited `Sprite` to be used as single objets and not references. Sprites are no longer stored into `std::unique_ptr` and are passed by copy. I think also for 2 strings it's not worth using `std::unique_ptr` and potentially struggle with references.
By copying, we don't use raw pointers from `.get()` anymore.